plonky2/src/gadgets/random_access.rs

```diff
@@ -2,15 +2,15 @@ use alloc::vec::Vec;
 
 use crate::field::extension::Extendable;
 use crate::gates::random_access::RandomAccessGate;
-use crate::hash::hash_types::RichField;
+use crate::hash::hash_types::{HashOutTarget, MerkleCapTarget, RichField};
 use crate::iop::ext_target::ExtensionTarget;
 use crate::iop::target::Target;
 use crate::plonk::circuit_builder::CircuitBuilder;
+use crate::plonk::circuit_data::VerifierCircuitTarget;
 use crate::util::log2_strict;
 
 impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
-    /// Checks that a `Target` matches a vector at a non-deterministic index.
-    /// Note: `access_index` is not range-checked.
```

Comment:
This was outdated, the current impl does effectively range check `access_index` by splitting it into bits (and requiring that the length is a power of two). Not sure if we want to guarantee that though. The old impl was lower-degree and could be useful at some point.